### DISCUSSION TITLE
I would like to compare to constants, not strings

### DISCUSSION THREAD

**Discussion by danehammer** [charmbracelet/bubbletea#697] [General]
```go
		switch keypress := msg.String(); keypress {
		case "ctrl+c":
```
This just feels a little dirty. I see there's constants in `key.go`, can we export them?

**Comment by hopefulTex** [comment]
The `key` example at the bottom shows how you can set and check keybindings.
You can use `key.Matches(msg, keybind)` for your cases.
[Check the bottom of the README](https://github.com/charmbracelet/bubbles/)

**Comment by meowgorithm** [comment]
Hi! There actually are exported constants for keys.

A `KeyMsg` is really just a `Key`, which looks like this:

```go
type Key struct {
	Type  KeyType
	Runes []rune
	Alt   bool
}
```

So here's how you could work with key constants:

```go
switch msg := msg.(type) {
case KeyMsg:
    switch msg.Type {
    case tea.KeyEnter:
        fmt.Println("you pressed enter!")
    case tea.KeyRunes:
        switch string(msg.Runes) {
        case "a":
            fmt.Println("you pressed a!")
        }
    }
}
```

Keep in mind there are no constants for letter keys (such as the "a" key) because we need to be able to match all sorts of input like `ö`, `ø`, and `猫`—hence `KeyRunes`. Additionally, Input method editors can input several runes at once (i.e. `猫咪`), hence the need for a slice.

Anyway, for more info see [Key](https://pkg.go.dev/github.com/charmbracelet/bubbletea#Key), [KeyMsg](https://pkg.go.dev/github.com/charmbracelet/bubbletea#KeyMsg), and [KeyType](https://pkg.go.dev/github.com/charmbracelet/bubbletea#KeyType) in the docs.
